python/sosmurf/StreamBase/_StreamBase.py

```diff
@@ -8,13 +8,16 @@ class StreamBase(pyrogue.Device):
     """
         Stream Base pyrogue object.
     """
-    def __init__(self, name, debug_data=False, debug_meta=False, debug_builder=False, agg_time=1.0, **kwargs):
+    def __init__(self, name, debug_data=False, debug_meta=False,
+                 debug_builder=False, agg_time=1.0, builder_encode=False,
+                 data_algo=3, tes_bias_algo=3, primary_algo=3, time_algo=3,
```

Comment:
One more comment ... I am not sure if the defaults for primary_algo, tes_bias_algo, and times_algo should be 2 or 3.  Those are largely noiseless fields so FLAC isn't going to help much.  But on the other hand if FLAC is better at saying "this is 1000 duplicates of the same value" than bzip is, you could save some time and space.  Either way should be fine.

Reply:
How does bzip handle linearly increasing data like timetsamps and frame-counters? I figured FLAC would be very good at that

Reply:
Fair point, it might do well.

But times_algo=3 should cause an error on serialization, because only 0 and 2 are supported.  Has this been tested?

Reply:
(FLAC is only applied to the lowest 24 bits... with a 10ns tick and 200 Hz sampling, the increment is 500000; that blows through FLAC's 24-bit range in ~34 steps.  So it probably wouldn't like that signal, and we'd be relying on bzip for the upper bytes anyway.)